src/components/action-bar/action-bar.spec.tsx

```diff
@@ -0,0 +1,184 @@
+import { h } from '@stencil/core';
+import { newSpecPage, SpecPage } from '@stencil/core/testing';
+import { ActionBarItem, ListSeparator } from '@limetech/lime-elements';
+import { ActionBar } from './action-bar';
+
+let page: SpecPage;
+let intersectionObserverMockedFunctions;
+let intersectionObserverMock;
+let intersectionCallback;
+const actionsIn: Array<ActionBarItem | ListSeparator> = [
+    {
+        text: 'Add',
+        icon: 'info',
+    },
+    {
+        text: 'Edit',
+        icon: 'idea',
+    },
+    {
+        text: 'Delete',
+        icon: 'multiply',
+    },
+    {
+        separator: true,
+    },
+    {
+        text: 'Settings',
+        icon: 'external_link',
+    },
+];
+describe('action-bar', () => {
+    beforeEach(async () => {
+        intersectionObserverMockedFunctions = {
+            observe: jest.fn(),
+            unobserve: jest.fn(),
+            disconnect: jest.fn(),
+        };
+        intersectionObserverMock = jest.fn((callback) => {
+            intersectionCallback = callback;
+
+            return intersectionObserverMockedFunctions;
+        }) as any;
+        global.IntersectionObserver = intersectionObserverMock;
+
+        page = await newSpecPage({
+            components: [ActionBar],
+            template: () => <limel-action-bar actions={actionsIn} />,
+        });
+
+        await page.waitForChanges();
+    });
+
+    it('renders', () => {
+        expect(page.root).toEqualHtml(`
+            <limel-action-bar>
+                <mock:shadow-root>
+                    <div class=items>
+                        <limel-action-bar-item isvisible=""></limel-action-bar-item>
+                        <limel-action-bar-item isvisible=""></limel-action-bar-item>
+                        <limel-action-bar-item isvisible=""></limel-action-bar-item>
+                        <limel-action-bar-item isvisible=""></limel-action-bar-item>
+                        <limel-action-bar-item isvisible=""></limel-action-bar-item>
+                    </div>
+                </mock:shadow-root>
+            </limel-action-bar>
+      `);
+    });
+
+    it('observes all actions in the intersection observer', () => {
+        expect(
+            intersectionObserverMockedFunctions.observe
+        ).toHaveBeenCalledTimes(5);
+    });
+
+    it('hides non-visable items in the menu', async () => {
+        triggerIntersection([true, true, false, false, false]);
+
+        await page.waitForChanges();
+        expect(getNumVisibleActions()).toBe(2);
+        expect(hasOverflowMenu()).toBeTruthy();
```

Comment:
This can be removed since it's part of the next test which has identical setup (or the other test can be removed :smile: )

Reply:
Yes, you are right.